tools/include/target.hpp

```diff
@@ -115,17 +116,17 @@ argument copy_from_target(T&, const argument& arg)
 }
 
 template <class T>
-float target_is_supported(T&, instruction_ref, support_metric)
+supported_segments target_find_supported(T&, const module*, support_metric)
 {
-    return 0;
+    return {};
 }
 
 <%
 interface('target',
      virtual('name', returns='std::string', const=True),
      virtual('get_passes', ctx='context&', options='const compile_options&', returns='std::vector<pass>', const=True),
      virtual('get_context', returns='context', const=True),
-     virtual('is_supported', returns='float', ins='instruction_ref', m='support_metric', const=True, default='target_is_supported'),
+     virtual('find_supported', returns='supported_segments', mod='const module*', m='support_metric', const=True, default='target_find_supported'),
```

Comment:
I would prefer `const module&` or `const_module_ref`(which is the same as `const module*`).